lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -1161,46 +1161,46 @@ def exec_module(self, **kwargs):
                     # os disk
                     if not vm_dict['properties']['storageProfile']['osDisk'].get('managedDisk'):
                         managed_disk = None
-                        vhd = self.compute_models.VirtualHardDisk(uri=vm_dict['properties']['storageProfile']['osDisk']['vhd']['uri'])
+                        vhd = self.compute_models.VirtualHardDisk(uri=vm_dict['properties']['storageProfile']['osDisk'].get('vhd', {}).get('uri'))
```

Comment:
Same question here- so this change keeps the module from bombing, but are we actually doing the correct thing? We're not in the managed disk case, so why do we have an OS disk without a vhd?

Reply:
agree with matt. the change itself looks good, but it doesn't resolve the issue #39958 because it doesn't resolve the key not exists, but change it to none error.

Reply:
@yungezz How to deal [#39958](https://github.com/ansible/ansible/issues/39958) and [#41003](https://github.com/ansible/ansible/issues/41003)? It seem fixed by this PR? Could we can close these two issue and create new PR to fix the key not exists issue ? Thanks!